defs.bzl

```diff
@@ -4,83 +4,73 @@ Binary buildifier / buildozer setup
 
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive", "http_file")
 load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
-
-_VERSION = "4.2.3"
-_BINARIES = [
-    # platform, arch, buildifier sha, buildozer sha
-    (
-        "darwin",
-        "amd64",
-        "954ec397089344b1564e45dc095e9331e121eb0f20e72032fcc8e94de78e5663",
-        "edcabae1d97bdc42559d7d1d65dfe7f8970db8d95d4bc9e7bf6656a9f2fb5592",
-    ),
-    (
-        "darwin",
-        "arm64",
-        "9434043897a3c3821fda87046918e5a6c4320d8352df700f62046744c4d168a3",
-        "f8d0994620dec1247328f13db1d434b6489dd007f8e9b961dbd9363bc6fe7071",
-    ),
-    (
-        "linux",
-        "amd64",
-        "a19126536bae9a3917a7fc4bdbbf0378371a1d1683ab2415857cf53bce9dee49",
-        "6b4177321b770fb788b618caa453d34561b8c05081ae8b27657e527c2a3b5d52",
-    ),
-]
+load("//:buildtools.bzl", "buildtools")
 
 def _buildifier_toolchain_setup_impl(repository_ctx):
-    content = 'load("@buildifier_prebuilt//:toolchain.bzl", "declare_toolchain")'
+    assets = buildtools.asset_from_json(repository_ctx.attr.assets_json)
 
-    for platform, arch, _, _ in _BINARIES:
-        name = "{}_{}".format(platform, arch)
+    content = 'load("@buildifier_prebuilt//:toolchain.bzl", "declare_toolchain")'
+    for asset in assets:
         content += """
 declare_toolchain(
-    name = "{name}",
-    buildifier = "@buildifier_{platform}_{arch}//file:buildifier",
-    buildozer = "@buildozer_{platform}_{arch}//file:buildozer",
+    tool_name = "{tool_name}",
+    tool = "@{uniq_name}//file:{tool_name}",
     os = "{platform}",
     arch = "{arch}",
 )
 
-""".format(name = name, platform = platform, arch = arch)
+""".format(
+            uniq_name = buildtools.create_unique_name(asset),
+            tool_name = asset.name,
+            platform = asset.platform,
+            arch = asset.arch,
+        )
     repository_ctx.file("BUILD.bazel", content)
 
 _buildifier_toolchain_setup = repository_rule(
     implementation = _buildifier_toolchain_setup_impl,
+    attrs = {
+        "assets_json": attr.string(
+            doc = "The assets to download encoded as JSON.",
+            mandatory = True,
+        ),
+    },
 )
 
-def buildifier_prebuilt_deps():
```

Comment:
We should probably leave this and call the other one to not break users?

Reply:
I moved this to `deps.bzl` due to dependencies. I added a dependency on `buildtools.bzl` at the top of this file. 
 The `buildtools.bzl` file has a dependency on `bazel-skylib`. The original version of `defs.bzl` contained the dependency declaration for `bazel-skylib`. So, I believe that there is no way to load a file that has a transitive dep on something that we have not declared yet. I could keep the dependency declaration in `defs.bzl` and then move the other logic to a new file. However, I selected this tact because it seemed to fit the Bazel rules pattern of dependencies being declared in `deps.bzl` and interesting logic declarations being in `defs.bzl`. Let me know if you think there is another way to tackle this.

Reply:
Ok yea seems fine, we'll just have to make sure to update the instructions when doing another release. I only did it in the same file before to keep user's WORKSPACES slightly cleaner with 1 less load